waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -276,3 +278,9 @@ genEnvValidationScript =
     [ C.mkTmplFd [relfile|scripts/validate-env.mjs|],
       C.mkUniversalTmplFdWithDst [relfile|validators.js|] [relfile|scripts/universal/validators.mjs|]
     ]
+
+genStaticAssetsDir :: AppSpec -> Generator [FileDraft]
+genStaticAssetsDir spec = return $ maybeToList maybeCopyStaticAssetsDir
+  where
+    maybeCopyStaticAssetsDir = createCopyDirWithMergeFileDraft staticAssetsDstDir <$> AS.staticClientAssetsDir spec
```

Comment:
> Why is it important for you to not delete the existing directory if it already exists? I would expect the opposite, that we want to delete it, to make sure we don't have any old files remaining there?

We create a default public folder with:
- `manifest.json`
- `favicon.ico`

and we reference those two files in `index.html`

If the user only added `favicon.ico` in their `public` folder -> `manifest.json` would be lost, but we could have saved it -> since it doesn't bother the user. I can't imagine the way it could be problematic. Therefore, I'm trying to keep the default files. Old files not being deleted ... yes, that's an issue, I'm not sure how to handle it.

Reply:
Ay yay, so we have our stuff there.

I still don't think it is ok to leave old stuff there -> user could put robots.txt in public, then remove it, but we still have it in generated project remaining from previous compilations, and that is an issue. It is a reason why we make sure with filedrafts that old files get removed from the generated project, if their filedraft doesn't exist any more.

What is then probably better is to add our own filedrafts for public/ -> manifest.json and favicon.ico.

I still think it is best to drive this through filedrafts if possible, because they have all the mechanisms we need in place, we don't need to worry about the state of the disk and similar, we just need to produce correct filedrafts and we are good.

Reply:
How would you do it? Go through the folder recursively and create file drafts explicitly instead of using `copyDir` file draft?

Reply:
Yeah, we already have function for it, `ExternalCode.readFiles`.

Check out what I wrote in this comment about possibly modifying it to provide mapping function for dstPath: https://github.com/wasp-lang/wasp/pull/1229#issuecomment-1574293126 .
I am talking a bit out of my butt here, but without investigating too much, it sounds like it could be the simplest solution (unless I forgot some caveats).